### DISCUSSION TITLE
q-tab-panel not active print Cannot read properties of undefined (reading '$refs')

### DISCUSSION THREAD

**Discussion by wiledd** [quasarframework/quasar#16329] [General - Components / Directives / etc] [answered]
Hello,
I have a form that works fine using validations using "this.$refs.rPCode.validate()"

However now I organize the components with q-tab and 3 q-tab-panel the command this.$refs.rPCode.validate()  generates "**Cannot read properties of undefined (reading '$refs')**" when the q-tab-panel containing the rPCode is not active (Principal).

When q-tab-panel containing rPCode is active it works fine.
[Resumen code.txt](https://github.com/quasarframework/quasar/files/12571424/Resumen.code.txt)

**Comment by metalsadman** [comment]
you have to make the tabs known, traversing it somewhere on mounted or wherever it makes sense using its exposed methods.
example https://codepen.io/metalsadman/pen/OJyoYPB

please paste the code instead of a file, using triple backticks for code formatting.

**Reply by wiledd** [reply, marked as answer]
thanks metasandman!!! I solved it using the @before-transition event, it's not a clean solution but it works :D

A pleasure and greetings from Peru

<q-tab-panels ref="tabRef" v-model="tab" animated class=""  **@before-transition**="tabChange">
